src/cache/mod.rs

```diff
@@ -317,6 +256,91 @@ impl Cache {
     }
 }
 
+fn send_updates(
+    peer_message: &PeerMessage,
+    peer_message_cache: &mut PeerMessageCache,
+    subscribers: &mut HashMap<Recipient<PeerDataArray>, PeerMessages>,
+) -> Vec<Recipient<PeerDataArray>> {
+    // Iterate through subscribers and send latest data based on their last_updated time
+    // Can be optimised for usual best case with fallback to binary_search
+    let mut dead = Vec::new();
+    let now = Instant::now();
+    for (recipient, peer_messages) in subscribers
+        .iter_mut()
+        .filter(|(_, s)| s.0.iter().find(|(p, _)| p == &peer_message).is_some())
+    {
+        if let Err(e) = update_subscriber(
+            recipient,
+            peer_messages,
+            peer_message,
+            &now,
+            peer_message_cache,
+        ) {
+            debug!("Unable to send PeerDataResponse: {:?}", e);
+            dead.push(recipient.to_owned());
+        }
+    }
+    dead
+}
+
+fn update_subscriber(
+    recipient: &Recipient<PeerDataArray>,
+    peer_messages: &mut PeerMessages,
+    peer_message: &PeerMessage,
+    now: &Instant,
+    peer_message_cache: &mut PeerMessageCache,
+) -> Result<(), SendError<PeerDataArray>> {
+    let update_time = peer_messages
+        .0
+        .get_mut(&peer_message)
+        .expect("Must not be modified anywhere else");
+    let idx = match peer_message_cache
+        .deque
+        .binary_search_by(|item| item.created_at.cmp(&update_time))
+    {
+        Ok(n) => Some(n + 1),
+        _ => None,
+    };
+    let idx = match idx {
+        Some(n) => n,
+        None => {
+            debug!("Finding closest time");
+            peer_message_cache
+                .deque
+                .iter_mut()
+                .enumerate()
+                .min_by(|a, b| {
+                    let y =
+                        a.1.created_at
+                            .signed_duration_since(*update_time)
+                            .num_milliseconds();
+                    let z =
+                        &b.1.created_at
+                            .signed_duration_since(*update_time)
+                            .num_milliseconds();
+                    (y * y).cmp(&(z * z))
+                })
+                .expect("")
+                .0
+        }
+    };
+    let response_data = peer_message_cache.deque[idx..].to_vec();
+    let pdr = PeerDataArray {
```

Comment:
Does `pdr` stand for `peer_data_response`? I for one prefer longer variable names.